db/column_family.cc

```diff
@@ -117,24 +117,28 @@ void GetIntTblPropCollectorFactory(
 }
 
 Status CheckCompressionSupported(const ColumnFamilyOptions& cf_options) {
-  if (!cf_options.compression_per_level.empty()) {
-    for (size_t level = 0; level < cf_options.compression_per_level.size();
-         ++level) {
-      if (!CompressionTypeSupported(cf_options.compression_per_level[level])) {
-        return Status::InvalidArgument(
-            "Compression type " +
-            CompressionTypeToString(cf_options.compression_per_level[level]) +
-            " is not linked with the binary.");
+  MutableCFOptions moptions(cf_options);
+  ImmutableCFOptions ioptions(cf_options);
+  if (moptions.compressor && !moptions.compressor->Supported()) {
+    return Status::InvalidArgument("Compression type " +
+                                   moptions.compressor->GetId() +
+                                   " is not linked with the binary.");
+  } else if (moptions.bottommost_compressor &&
+             !moptions.bottommost_compressor->Supported()) {
+    return Status::InvalidArgument("Compression type " +
+                                   moptions.bottommost_compressor->GetId() +
+                                   " is not linked with the binary.");
+  } else if (!moptions.compressor_per_level.empty()) {
+    for (const auto& compressor : moptions.compressor_per_level) {
+      if (!compressor->Supported()) {
```

Comment:
Is there any possibility that this compressor is null?

Reply:
- compressor_per_level is initialized as an empty vector. 
- MutableCFOptions::RefreshDerivedOptions populates compressor_per_level using the content of compression_per_level.
- Compressors are obtained from BuiltinCompressor::GetCompressor, which could return nullptr in case of an invalid selection in compression_per_level.

I thought an invalid selection in compression_per_level would be caught before reaching this method, but that's incorrect. I added a null check and a test in compression_test (to verify the expected error is returned in case of invalid selection in compression_per_level).